capsules/extra/src/ieee802154/framer.rs

```diff
@@ -450,11 +441,18 @@ impl<'a, M: Mac<'a>, A: AES128CCM<'a>> Framer<'a, M, A> {
                         // specifies `KeyIdMode::Source4Index`, the source
                         // address used for the nonce is actually a constant
                         // defined in their spec
-                        let device_addr = match self.lookup_addr_long(header.src_addr) {
-                            Some(addr) => addr,
+                        let device_addr = match header.src_addr {
+                            Some(mac) => match mac {
+                                MacAddress::Long(val) => val,
+                                MacAddress::Short(_) => {
+                                    kernel::debug!("[15.4] DROPPED PACKET - error only short address provided on encrypted packet.");
+                                    return None
+                                },
+                            },
                             None => {
-                                return None;
-                            }
+                                kernel::debug!("[15.4] DROPPED PACKET - Malformed, no src address provided.");
+                                return None
+                            },
```

Comment:
@amit or @hudson-ayers or maybe @bradjc: Do we have a policy in Tock about "permanent" debug statements in capsules?

I think I'm generally against them, but if other capsules are doing it anyways, it doesn't hurt here.